src/Panel/Panel.php

```diff
@@ -446,9 +446,14 @@ public static function routesForSearches(string $areaId, array $area): array
 				'type'    => 'search',
 				'area'    => $areaId,
 				'action'  => function () use ($params) {
-					$request = App::instance()->request();
-
-					return $params['query']($request->get('query'));
+					$kirby   = App::instance();
+					$request = $kirby->request();
+					$query   = $request->get('query');
+					$limit   = (int)$request->get('limit', 10);
+					$limit ??= $kirby->option('panel.search.limit', 10);
```

Comment:
I do not understand these lines. First line has fallback set, second line uses `??=` 🤔 

Also I couldn't get the new `panel.search.limit` setting to work.

Reply:
Oh I think the `, 10` shouldn't be there on line 452.

Reply:
No, also casting prevents to running next line. I think `(int)null` means `0`.

Reply:
True. Ok I tried to format it nicely, but then it's better to make it a one liner:
```
$limit  = (int)$request->get('limit', $kirby->option('panel.search.limit', 10));
```

Reply:
Also one more thing; `$request->get('limit')` return always `15`. This prevents running fallback option. Do you know why?

````
/panel/search?type=pages&query=test
``

Reply:
For the search view that's correct. The option is only meant for the dialog (as it has been in the past)